src/GitHub.App/Services/PullRequestService.cs

```diff
@@ -89,7 +89,22 @@ public IObservable<string> GetPullRequestTemplate(ILocalRepositoryModel reposito
         public IObservable<bool> IsWorkingDirectoryClean(ILocalRepositoryModel repository)
         {
             var repo = gitService.GetRepository(repository.LocalPath);
-            return Observable.Return(!repo.RetrieveStatus().IsDirty);
+            var isClean = !IsFilthy(repo.RetrieveStatus());
+            return Observable.Return(isClean);
+        }
+
+        static bool IsFilthy(RepositoryStatus status)
```

Comment:
Is this actually needed though? Can't we just let libgit2sharp error out if it doesn't manage the operation? I honestly can't remember why I felt it necessary to do a check for cleanliness here :/

Also LOL at the name ;)

Reply:
> Is this actually needed though? Can't we just let libgit2sharp error out if it doesn't manage the operation?

I wanted to start with a simple drop-in replacement for what we've been using (a simple bug fix). Letting libgit2sharp error out might be better and give the user more info, but would also require more testing.

> I honestly can't remember why I felt it necessary to do a check for cleanliness here :/

I know command line Git lets you checkout when there modified files, as long as they don't conflict with changes in the branch you're checking out. I'm not sure this is something we would want to allow in this context. Maybe this is an option, I'll investigate.

If you're happy with this simple fix, I'd prefer to start with that and improve later. Can I open a separate PR that uses the letting libgit2sharp error out technique?

Reply:
Here is a PR that lets libgit2sharp error out rather than simply allowing un-tracked files: #1304. I'm going to document the implications of this further and get more eyes on it.

Until #1304 is approved, I'm keen to get this PR merged as a point fix. I don't mind reverting it later, but I'm sick of not being able to check-out PRs! 😉  

Does that make sense?